integration/tests/epochs/suite.go

```diff
@@ -487,14 +487,18 @@ func (s *Suite) ExecuteGetProposedTableScript(ctx context.Context, env templates
 	return v
 }
 
-// SubmitSetApprovedListTx adds a node the the approved node list, this must be done when a node joins the protocol during the epoch staking phase
+// SubmitSetApprovedListTx adds a node to the approved node list, this must be done when a node joins the protocol during the epoch staking phase
 func (s *Suite) SubmitSetApprovedListTx(ctx context.Context, env templates.Environment, identities ...flow.Identifier) *sdk.TransactionResult {
-	ids := make([]cadence.Value, 0)
+	cdcApprovedNodeDictEntries := make([]cadence.KeyValuePair, 0, len(identities))
 	for _, id := range identities {
-		idCDC, err := cadence.NewString(id.String())
+		cdcNodeID, err := cadence.NewString(id.String())
 		require.NoError(s.T(), err)
 
-		ids = append(ids, idCDC)
+		kvPair := cadence.KeyValuePair{
+			Key:   cdcNodeID,
+			Value: cadence.NewBool(true),
+		}
+		cdcApprovedNodeDictEntries = append(cdcApprovedNodeDictEntries, kvPair)
 	}
```

Comment:
Looks similar to the code in `fvm/blueprints/epochs.go`. I know this is a test, but It might still make sense extracting that into a function.

Reply:
https://github.com/onflow/flow-go/pull/3837/commits/42589d24916919e7f22c39830db1e9c34a95ed7d